### DISCUSSION TITLE
why use "Simple Addition" & "Booth function" in example file

### DISCUSSION THREAD

**Discussion by seghier** [Tunny-gh/Tunny_in_Open_Sea#55] [General]
Hello
Can you please show a video tutorial with step by step for the example included with the plugin and some other examples.
I don't understand why you used this function and why you used addition

![image](https://user-images.githubusercontent.com/6026588/173753502-387caf60-9f49-4d23-b347-f363afb75845.png)

Also when i use the length of the curve and the volume as objectives , it show one result when i click Restore than it show different results when i click Reflect, i don't understand what this terms mean (restore and reflect), i hope you make them more clear.
The optimization stop working when i test the sample file with different settings even i clear the results

**Comment by hrntsm** [comment]
> I don't understand why you used this function and why you used addition

The Booth function and Addition components included in the sample are not meaningful. I use this as one example to show that Tunny can take multiple objective functions.

> Also when i use the length of the curve and the volume as objectives , it show one result when i click Restore than it show different results when i click Reflect, i don't understand what this terms mean (restore and reflect), i hope you make them more clear.

Sorry I don't have a clear explanation of restore and reflect.
The two words function as follows

- Set restore model number input and clicking **the Restore button** is restored from the optimization results file and is the output of the component.
   - The model number matches the tree structure of the output.
   - -1 is input, the results of all models on the Pareto front will output.
   - -10 is input, the results of all models will output
- Set restore model number input and Clicking **the Reflect button** will cause Grasshopper to reflect the results to number slider and genepool.
   - Only one model number can be entered when using this feature

> The optimization stop working when i test the sample file with different settings even i clear the results

This doesn't help me understand what is causing the problem, so please give me details on what condition it is in and what errors are being output.
Images would also be helpful.

**Comment by seghier** [comment]
Thanks, this is a recorded video
https://www.dropbox.com/s/no390bq7q15uny3/TUNNY.mp4
You can see that the texts are not aligned, and how we know the best results in this case (minimum length & minimum volume), some results are very different?
`Sorry I don't have a clear explanation of restore and reflect.` : What i mean is using well-known terms like maybe Get data , update data ... etc
**Suggestions:**
1- Adding output for true/false indicate when optimization start or stop
2- Maybe change RunOptimize to start?
3- Make inputs accept lists (Merge)
![image](https://user-images.githubusercontent.com/6026588/173844101-ebc6d034-314e-4732-b331-03e71c61d757.png)
4- Add option to choose between: maximize, minimize or target value (like in Galapagos)

And thank you very much for the nice features like saving datas and geometries, this is very useful

**Reply by hrntsm** [reply]
Thanks for the suggestion.

1. Adding output for true/false indicate when optimization start or stop
    - I will put it on the development schedule. 
1. Maybe change RunOptimize to start?
    - I'm not sure what this means.
1. Make inputs accept lists (Merge)
    - No plans for implementation, as it makes sense to point to the target component itself.
1. Add option to choose between: maximize, minimize or target value (like in Galapagos)
    - If you want to maximize the value, you can simply reverse the positive and negative values using the negative component, so there are no plans to implement maximization.

**Comment by phm27** [comment]
Hello, I'm trying to get back the output after an optimization, but I have an error message :  "There are no output models. Please check the stuty name."
Can you help me?

**Reply by hrntsm** [reply]
Hi @phm27 !

The plotting of results is working fine, but only the Output to Grasshopper is not working?
If the plot does not work either, the results may not be saved properly.

The optimization result is created in the same folder as the Tunny.gha file under the name "Tunny_Opt_Result.db". Is this file created?
<img width="313" alt="image" src="https://user-images.githubusercontent.com/23289252/189051285-21f7ac1e-f5d9-4be0-ad7a-ec6d8edadc9f.png">

**Reply by phm27** [reply]
Hi,
I confirm that the plot is working and the Tunny_Opt_Result.db is generated.

**Reply by hrntsm** [reply]
Which button are you trying to output?
If it is a single-objective optimization rather than a multi-objective optimization, using ParetoSolutions does not work and show error message "There are no output models. Please check the study name."
Please use "All trials" or "Use model number" to output your results.

**Reply by phm27** [reply]
Hi, the error message is no longer displayed, but I'm not able to get back the output. Nothing happen when I press "Pareto Solution"  and the process goes on endless when I press All Trials.
Maybe I have a mistake in my model?


![captureTunny](https://user-images.githubusercontent.com/113048949/189177567-f9e98929-e6f1-46b3-9e2d-b489a8589f60.png)

**Reply by hrntsm** [reply]
If the progress bar does not reach 100% and the last little bit remains, the window that tells you that the output is complete may be hidden in the Rhino window.
May I ask you to check if that window is not showing up?

<img width="844" alt="Screen Shot 2022-09-09 at 17 42 54" src="https://user-images.githubusercontent.com/23289252/189309805-1b535daa-897a-4609-8fd9-8123521d6d00.png">

If it still does not work, it may not be good that all the sliders connected to Variable have the same name. Could you try with unique values?

I will check it out as it may be a bug that it does not do anything when ParetoSolution is selected.

**Reply by phm27** [reply]
Unfortunatly the windows isn't showing up.
I tried to change the sliders names, but I still have the same probleme, the process goes on endless

**Reply by hrntsm** [reply]
@phm27 
I created a GH file with the same contents as yours and tried it on my machine and got the same phenomenon as yours.
It was a bug in the software and I will investigate the cause. Please wait a little.

**Reply by hrntsm** [reply]
The only supported Geometry types are "Curve", "Brep", "Mesh", "Surface", and "SubD", but the "Circle" input was the bug that was causing the problem.
In the next version, I' ll try to make the other items as well, so please use the Curve component as shown in the following image.

<img width="862" alt="image" src="https://user-images.githubusercontent.com/23289252/189648148-2cb82a26-6721-4c1c-a22a-a9c780d485fb.png">

**Reply by phm27** [reply]
Hi, it works perfectly. Many thanks for this and for the work.
Any chance to have tunny on mac?

**Reply by hrntsm** [reply]
Unfortunately, the library used in Tunny does not work on mac, so there are no plans to support mac.

**Comment by sperrault** [comment]
Hi ! 
I can't launch the solver; the canvas blinks and nothing happens. Any idea ?
![Tunny_issue](https://user-images.githubusercontent.com/11261470/199065083-36fd9b6b-660f-47dd-8ae4-251f3be4c862.gif)

**Reply by hrntsm** [reply]
Hi! @sperrault 

Tunny downloads some files automatically, does the folder with Tunny.gha have the following file structure?
If not, the automatic download may have failed.

![Screenshot 2022-11-01 at 11 48 25](https://user-images.githubusercontent.com/23289252/199147810-e06a5e6e-1492-4cf8-bd45-7286de7c75e2.png)

**Reply by sperrault** [reply]
Thanks @hrntsm, I have this file structure.
Is there any known compatibility issue with other plugins ?

**Reply by hrntsm** [reply]
Thanks to reply @sperrault 

Here are some solutions that come to mind.
- If Wallacei is included, this may cause problems.
- The others are basic, but it is possible that you forgot to make the downloaded ZIP no "blocked" before installing.
- Since you are touching the sample file, I assume there is no problem, but if some of the inputs to the Tunny component are null, you will get this kind of error.

**Reply by sperrault** [reply]
Hi @hrntsm, 
- I removed Wallacei, and all other components
- All files were unblocked
- Work on unmodified sample file, no null inputs
Also : 
- tried opening rhino as administrator
- tried moving component folder to C:\Tunny
- changed Python environment variables paths..
Without success..
Thank you !

**Reply by hrntsm** [reply]
Hi @sperrault ,

The only other thing I can think of is that Tunny downloads the Python libraries behind Grasshopper, but if your PC has tight security, that download may fail.
Sorry it didn't work while you tried everything. 🙇 
I' m working on developing it to be better!
